### DISCUSSION TITLE
Exporting transient displacement coordinates of nodes

### DISCUSSION THREAD

**Discussion by felixschurk** [ansys/pydpf-post#751] [General]
Hei, 
I am currently running a transient structural analysis in Ansys and want to perform some post processing with pydpf-post. 
As the simulation is fairly detailed, the `.rst` file is around 22 GB.

My main problem now is the runtime of the Python script, loading the data, exporting to displacement is still reasonable from the runtime. However, then exporting the coordinates from a specific node over all time steps is really really really slow (talking here about 1h and more, just for the export).

Based on the documentation (and try outs) it is not currently supported to extract all displacement data in a coordinate, from a node at once it is only doable with a loop. At least the `array` can only do it for a simple node.
Attached is a snipped providing my current workaround
```python
displacement_zy_421 = np.zeros([max(simulation.set_ids), 2])
selected_node_id = 1569

for i,val in enumerate(simulation.set_ids):
    displacement_zy_421[i,0] = displacement.select(
            set_ids=[val],
            node_ids=selected_node_id,
            components=["Z"]
        ).array
    
    displacement_zy_421[i,1] = displacement.select(
            set_ids=[val],
            node_ids=selected_node_id,
            components=["Y"]
        ).array
```
Maybe only having one call to the displacement and directly extracting `Y` and `Z` in one go improves it a bit, I will try it out and report. 

Okay at least for a small sample set it was quite as fast, but still in general slow.

```
displacement_zy_421 = np.zeros([max(simulation.set_ids), 2])
selected_node_id = 1569

for i,val in enumerate(simulation.set_ids):
    displacement_zy_421[i] = displacement.select(
            set_ids=[val],
            node_ids=selected_node_id,
            components=["Z", "Y"]
        ).array
```

But in general my question is, is there any method - or way - to easily export a whole time series of a point of interest?

**Comment by PProfizi** [comment]
Hi @felixschurk,
The [TransientMechanicalSimulation.displacement](https://post.docs.pyansys.com/version/stable/api/ansys.dpf.post.transient_mechanical_simulation.html#ansys.dpf.post.transient_mechanical_simulation.TransientMechanicalSimulation.displacement) method and all other result methods take in a `node_ids` argument, as well as an `all_sets` argument.
I think in your case what you want is to directly limit result reading to a single node and set `all_sets=True`.
This will give you the displacement at a given node for all time steps.

Is that what you were looking for?
